pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -122,6 +123,9 @@ public BaseBrokerRequestHandler(Configuration config, RoutingTable routingTable,
     } else {
       _tableCache = null;
     }
+
+    PinotQueryParserFactory.MAX_QUERY_SELECTION_LIMIT = _config.getInt(CommonConstants.Helix.MAX_QUERY_SELECTION_LIMIT_KEY, CommonConstants.Helix.DEFAULT_MAX_QUERY_SELECTION_LIMIT);
```

Comment:
we dont need to set the variable in PinotQueryParserFactory. Just change the brokerrequest using the config